R/transformCounts.R

```diff
@@ -477,63 +479,64 @@ setMethod("relAbundanceCounts",signature = c(x = "SummarizedExperiment"),
     return(mat)
 }
 
-#' @importFrom DelayedMatrixStats colSums2 colMeans2
-.calc_clr <- function(mat, ...){
-    # If there is negative values, gives an error.
-    if(any(mat <= 0, na.rm = TRUE)) {
-        stop("Abundance table contains zero or negative values and ",
-             "log10 transformation is being applied without pseudocount.\n",
-             "Try to add pseudocount (default choice pseudocount = 1 for count ",
-             "assay; or pseudocount = min(x[x>0]) for relabundance assay).",
-             call. = FALSE)
-    }
+#' @importFrom DelayedMatrixStats colSums2
+.calc_clr <- function(mat, method = "clr", pseudocount = NULL, ...){
     # Calculate colSums
     colsums <- colSums2(mat, na.rm = TRUE)
     # Check that they are equal; affects the result of CLR. CLR expects a fixed
     # constant
     if( round(max(colsums)-min(colsums), 3) != 0  ){
-        warning("All the total abundances of samples do not sum-up to a fixed constant. ",
+        warning("All the total abundances of samples do not sum-up to a fixed constant.\n",
```

Comment:
not clear what this test does; consider adding a test function that tests a specific thing.

Reply:
This is just a note for user; CLR is should be applied to values that sum-up to a fixed constant (e.g., relative abundances)

This might be something that user should know / this warning can be removed

Reply:
Warning is good to keep. I was just wondering if the round(..., 3) is an ideal way to test this. Perhaps it works for now if there is no obvious solution.

Reply:
Ahhh ok, when I created the warning message, I found out that "all colsums are equal" was too strict

Reply:
One potentially more clear way to do this might be:
`all.equal(x,y, tolerance=0.01)`

or 
`abs(x-y) < tolerance`